sdk/lib/opentelemetry/sdk/trace/export/batch_span_processor.rb

```diff
@@ -101,14 +106,21 @@ def work
           def export_batch(batch)
             result_code = nil
             @export_attempts.times do |attempts|
-              result_code = @exporter.export(batch)
+              result_code = export_with_timeout(batch)
               break unless result_code == FAILED_RETRYABLE
 
               sleep(0.1 * attempts)
             end
             report_result(result_code, batch)
           end
 
+          def export_with_timeout(batch)
+            Timeout.timeout(export_timeout_seconds) { @exporter.export(batch) }
+          rescue Timeout::Error
+            # TODO: tell exporter that it timed out? (e.g., cleanup network connections)
```

Comment:
The exporter should be able to `rescue Timeout::Error` and do its own cleanup. We don't need an extra cleanup API.

Reply:
I've removed the comment. :heavy_check_mark: